lms/djangoapps/instructor_task/tasks_helper.py

```diff
@@ -779,9 +779,27 @@ def upload_grades_csv(_xmodule_instance_args, _entry_id, course_id, _task_input,
 
         if not course_grade:
             # An empty gradeset means we failed to grade a student.
-            task_progress.failed += 1
-            err_rows.append([student.id, student.username, err_msg])
-            continue
+            # TODO: create a separate track report so that instructors
+            # do not need to use grade reports to view learners' track
+            # before the course starts. We special-case for this scenario
+            # here... for now.
+            if err_msg == "User does not have access to this course" and (
+                            course.start > datetime.now().replace(tzinfo=UTC) or not course.start is None
+            ):
```

Comment:
If we decide that `iter` is going to return a grade of 0 for users who do not have access to they course, why wouldn't we just decide that instead of raising a `PermissionDenied` when calling `create` for a user without access to the course, we will return a grade of 0 there? It seems inconsistent to return one thing in the bulk-create case, and another in the single-create case. 

Separately- as far as I know, the current course grade code meets our needs in every known case except for this edge case in which course teams are using grade reports for something that in reality has little-to-nothing to do with grades. Changing our core grading code for that case doesn't feel right when we are uncomfortable with course teams using the grade report this way anyway.

[Looking at this CCX code](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/ccx/views.py#L572), I think returning a grade of 0% instead of `None` in `iter` could also change the behavior of that report in an unexpected way.

So I'm going to propose a different approach- for the short term, we use the fix I've implemented here. Long-term (not that long, though- maybe in a couple sprints?), we update course grade access control to allow staff to view actual course grades (not 0's, whatever is in the database) for anyone enrolled in the course, regardless of the student's ability to see the course at that moment. Thoughts? Does that approach to course grade access for staff run counter to an established contract that the rest of the code expects? cc @jcdyer